PDO/PDO.php

```diff
@@ -1003,6 +1005,30 @@ public function exec ($statement) {}
 	 */
 	public function query ($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, $arg3 = null, array $ctorargs = array()) {}
 
+	#[PhpStormStubsElementAvailable('8.0')]
+	/**
+	 * (PHP 5 &gt;= 5.1.0, PHP 7, PECL pdo &gt;= 0.2.0)<br/>
+	 * Executes an SQL statement, returning a result set as a PDOStatement object
+	 * @link https://php.net/manual/en/pdo.query.php
+	 * @param string $statement <p>
+	 * The SQL statement to prepare and execute.
+	 * </p>
+	 * <p>
+	 * Data inside the query should be properly escaped.
+	 * </p>
+	 * @param int $mode <p>
+	 * The fetch mode must be one of the PDO::FETCH_* constants.
+	 * </p>
+	 * @param mixed $fetch_mode_args <p>
+	 * Arguments of custom class constructor when the <i>mode</i>
+	 * parameter is set to <b>PDO::FETCH_CLASS</b>.
+	 * </p>
+	 * @return PDOStatement|false <b>PDO::query</b> returns a PDOStatement object, or <b>FALSE</b>
+	 * on failure.
+	 * @see PDOStatement::setFetchMode For a full description of the second and following parameters.
+	 */
+	public function query ($statement, $mode = PDO::ATTR_DEFAULT_FETCH_MODE, ...$fetch_mode_args) {}
```

Comment:
May be it makes sense to add property and return types (including union types) directly to signatures for entities marked with `#[PhpStormStubsElementAvailable('8.0')]` ? Thus we will strict types and improve type inference.